tidy3d/components/source.py

```diff
@@ -796,10 +796,9 @@ def _tangential_component_defined(cls, val: FieldDataset, values: dict) -> Field
     def _check_fields_interpolate(cls, val: FieldDataset) -> FieldDataset:
         """Checks whether the filds in 'field_dataset' can be interpolated."""
         if isinstance(val, FieldDataset):
-            fields = ["Ex", "Ey", "Ez", "Hx", "Hy", "Hz"]
-            for field in fields:
-                if isinstance(val.__getattribute__(field), ScalarFieldDataArray):
-                    val.__getattribute__(field)._interp_validator()
+            for _, data in val.field_components.items():
+                if isinstance(data, ScalarFieldDataArray):
```

Comment:
huh, for some reason I didnt see that at all. Yea that should do it!

Reply:
only thing I would consider adding to make it just a bit more user friendly would be an optional keyword argument to `interp_validator(self, field_name: str=None)`

and then we can pass `permittivity` or `Ex`, etc and the error message can include it.

This might help, eg. users identify which field is causing issues.

If this overly complicates things, feel free to ignore.

Reply:
> If this overly complicates things, feel free to ignore.

Not at all! I think it'll be much nicer

Reply:
Looks good! minus a missing `f` in the fstring ;). Thanks!